### PR TITLE
Test and production deployments to PyPI from Github Actions

### PR DESCRIPTION
**Pull request by cdecker**
Totally overengineered answer to #4469.

 - [x] Need to latch onto the version being tagged and then publishing to PyPI
Changelog-None

### CODE REVIEW

**Comment by cdecker** [comment]
The events still need to be testing, but that'll obviously have to wait until we merge this. I think the triggers are set up correctly, but will monitor this during the release process.

File: .github/workflows/pypi.yml
Comment (vincenzopalazzo) [inline .github/workflows/pypi.yml:7]: Do you think that maybe it is better to use a label checker here? like this one? https://github.com/marketplace/actions/label-checker-for-pull-requests

We can make only push if there is some change in the pyln modules?

**Review by vincenzopalazzo — COMMENTED** [review]
LGTM https://github.com/ElementsProject/lightning/pull/4809/commits/5a449836ce0531d4ec717dd4e21c7c318d2ecb06

There is a recursion inside the[ dependencies resolution](https://github.com/ElementsProject/lightning/runs/3688356495?check_suite_focus=true) maybe this is not a problem because we need to release before some package?

or `pyln.proto>=0.9.3` will cause this recursion?

**Review by vincenzopalazzo — CHANGES_REQUESTED** [review]
I don't know why I can not put the comment inside the change, BTW I think that the version of the package `pyln.proto` could have a minimum requirement version like `pyln.proto>= 0.9.1`in the bolt7 (`contrib/pyln-spec/bolt7/requirements.txt`)

Or there is some reason that I'm missing here?

**Comment by cdecker** [comment]
> I don't know why I can not put the comment inside the change, BTW I think that the version of the package `pyln.proto` could have a minimum requirement version like `pyln.proto>= 0.9.1`in the bolt7 (`contrib/pyln-spec/bolt7/requirements.txt`)
> 
> Or there is some reason that I'm missing here?

Had to remove the minimum version temporarily due to the way we auto-generate the version in shallow git clones we would get v0.0.0.post10000 as pyln-bolt7 version thus failing the version check. I have since started checking out the tags to generate the version correctly

**Comment by cdecker** [comment]
ACK https://github.com/ElementsProject/lightning/pull/4809/commits/adfbb33fa24d99ffbb4508c591e2191be2d51a8b

**Pull request merged**
